src/Sarif/Readers/PropertyBagConverter.cs

```diff
@@ -49,6 +49,12 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                     serializedValue = JsonConvert.ToString(serializedValue);
                     isString = true;
                 }
+                else if (propertyType == typeof(DateTime))
```

Comment:
>else [](start = 16, length = 4)

The fix. A simple `else` might also work but this is the minimal change that fixes the bug.